Pulp tls update

jackhodgkiss opened the pull request.

Alex-Welsh commented on etc/kayobe/docker.yml:
```diff
@@ -30,7 +30,7 @@ docker_registry: "{{ stackhpc_docker_registry }}"
 
 # CA of docker registry
-#docker_registry_ca:
+docker_registry_ca: "{{ kayobe_env_config_path ~ '/openbao/OS-TLS-INT.crt' if pulp_enable_tls | bool else '' }}"
```
Lets not default to Bao in this instance. We still have many more people using vault on Caracal

jovial commented:
Would be good to get this one in. I guess it needs to retarget 2025.1 and switch to the new playbook directory structure, but overall looks good to me.

jackhodgkiss committed:
feat: add support for enabling `Pulp` TLS

Add playbooks, configuration and documentation to support the deployment
of Pulp TLS.

Alex-Welsh commented on etc/kayobe/ansible/deployment/copy-ca-to-hosts.yml:
```diff
@@ -0,0 +1,47 @@
+    # Work around no known_hosts entry on first boot.
+    ansible_ssh_common_args: -o StrictHostKeyChecking=no
+    # Don't assume facts are present.
+    os_family: "{{ ansible_facts.os_family | default('Debian' if os_distribution == 'ubuntu' else 'RedHat') }}"
```
The playbook isn't gathering facts, so why not just use os_distribution as the identifier throughout?

jackhodgkiss replied:
This convention is used elsewhere https://github.com/stackhpc/stackhpc-kayobe-config/commit/817bea96fc73097ec141a98ddc7c5da1ae39a1d4

Alex-Welsh commented on etc/kayobe/ansible/pulp/pulp-generate-certificate.yml:
```diff
@@ -0,0 +1,49 @@
+        file: "{{ kayobe_env_config_path }}/openbao/seed-openbao-keys.json"
+        name: openbao_keys
+
+    - name: Issue a certificate Pulp
```
```suggestion
    - name: Issue a certificate for Pulp
```
or
```suggestion
    - name: Issue a Pulp certificate
```

Alex-Welsh commented on etc/kayobe/ansible/pulp/pulp-generate-certificate.yml:
```diff
@@ -0,0 +1,49 @@
+          ip_sans: "{{ admin_oc_net_name | net_ip(inventory_hostname=groups['seed'][0]) }}"
+      register: pulp_certificate
+
+    - name: Ensure pulp certificates directory exists
```
```suggestion
    - name: Ensure Pulp certificates directory exists
```

Alex-Welsh commented on releasenotes/notes/pulp-tls-105e47f0da602a25.yaml:
```diff
@@ -0,0 +1,5 @@
+features:
+  - |
+    Add playbooks and configuration to enable the easy deployment of Pulp with
+    TLS support.
```
Could you also mention this only works with openbao?

Alex-Welsh commented on etc/kayobe/ansible/pulp/pulp-generate-certificate.yml:
```diff
@@ -0,0 +1,49 @@
+  vars:
+    openbao_api_addr: http://127.0.0.1:8200
+    openbao_intermediate_ca_name: OS-TLS-INT
+  tasks:
```
Worth adding a task at the start to just assert `stackhpc_ca_secret_store == openbao`

Alex-Welsh commented on doc/source/configuration/openbao.rst:
```diff
@@ -460,6 +460,40 @@ Enable the required TLS variables in kayobe and kolla
+
+.. warning::
+
+   For clouds in production consider the impact of enabling TLS on specific hosts as Docker daemon will be restarted and this will disrupt deployments of Ceph Reef and older.
```
We should mention it doesn't work for Vault

Alex-Welsh commented on doc/source/configuration/openbao.rst:
```diff
@@ -460,6 +460,40 @@ Enable the required TLS variables in kayobe and kolla
+   .. code-block::
+
+      # Whether to enable TLS for Pulp.
+      pulp_enable_tls: true
```
(opinion) you can make nice captions in sphinx, which looks good for the name of the file that is being edited
```suggestion
3. Enable TLS for Pulp

   .. code-block::
      :caption: $KAYOBE_CONFIG_PATH/pulp.yml

      # Whether to enable TLS for Pulp.
      pulp_enable_tls: true
```

jackhodgkiss committed:
docs: add suggested changes

- Clean up name of tasks
- Add OpenBao assertion
- Update release note

Signed-off-by: Jack Hodgkiss <jack@stackhpc.com>

jackhodgkiss commented:
This will require https://review.opendev.org/c/openstack/ansible-collection-kolla/+/955837/1/releasenotes/notes/improve-container-registry-ca-conditional-0247b0014379ba82.yaml due to https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/19670310731/job/56337904934#step:25:1902.

Backport upstream was rejected maybe downstream backport.

jackhodgkiss committed:
fix: remove `docker_registry_ca` from being set

The task that uses this variable cannot handle the variable being
empty or omitted. This has been left unset within the base environment.

Signed-off-by: Jack Hodgkiss <jack@stackhpc.com>

jackhodgkiss committed:
Merge branch 'stackhpc/2025.1' into pulp-tls-update

technowhizz commented:
is this ready to test elsewhere @jackhodgkiss

jackhodgkiss commented:
> is this ready to test elsewhere @jackhodgkiss 

Yeah if you have somewhere to test it. I have applied at some new deployments.

Alex-Welsh commented:
/gemini review

Alex-Welsh commented on etc/kayobe/ansible/deployment/copy-ca-to-hosts.yml:
```diff
@@ -0,0 +1,47 @@
+    # Work around no known_hosts entry on first boot.
+    ansible_ssh_common_args: -o StrictHostKeyChecking=no
+    # Don't assume facts are present.
+    os_family: "{{ ansible_facts.os_family | default('Debian' if os_distribution == 'ubuntu' else 'RedHat') }}"
```
Seems reasonable to leave in. Might run as a hook or called in after facts have already been gathered
